src/Payum/Core/PayumBuilder.php

```diff
@@ -373,6 +380,18 @@ public function setHttpClient(HttpClientInterface $httpClient = null)
     }
 
     /**
+     * @param MessageFactory $httpMessageFactory
+     *
+     * @return PayumBuilder
+     */
+    public function setHttpMessageFactory(MessageFactory $httpMessageFactory)
+    {
+        $this->httpMessageFactory = $httpMessageFactory;
+
+        return $this;
```

Comment:
No need for this I believe. If user want s to overwrite the default http client he can do it via `PayumBuilder:: addCoreGatewayFactoryConfig()` method. I am going to remove similar method `PayumBuilder::setHttpClient()` too.